web/src/compose_tooltips.ts

```diff
@@ -35,6 +34,50 @@ export function initialize(): void {
             instance.destroy();
         },
     });
+    delegate("body", {
+        target: "#compose_buttons .compose-reply-button-wrapper",
+        delay: EXTRA_LONG_HOVER_DELAY,
+        // Only show on mouseenter since for spectators, clicking on these
+        // buttons opens login modal, and Micromodal returns focus to the
+        // trigger after it closes, which results in tooltip being displayed.
+        trigger: "mouseenter",
+        appendTo: () => document.body,
+        onShow(instance) {
+            const $elem = $(instance.reference);
+            const button_type = $elem.attr("data-reply-button-type");
+            switch (button_type) {
+                case "disabled": {
+                    instance.setContent(
+                        parse_html($("#compose_reply_button_disabled_tooltip_template").html()),
```

Comment:
Let's rename `compose_reply_button_disabled_tooltip_template` to something like `compose_reply_direct_disabled_button_tooltip_template`, which is more verbose about the intent.

Reply:
Also, please add that as separate commit.